lib/core/src/test/java/io/atlasmap/core/DefaultAtlasExpressionProcessorTest.java

```diff
@@ -103,7 +103,7 @@ public void testFilter() throws Exception {
         assertEquals(1, childFieldGroup.getField().size());
         assertEquals("foo0", childFieldGroup.getField().get(0).getValue());
         childFieldGroup = (FieldGroup) fieldGroup.getField().get(1);
-        assertEquals("/testPathfoo<2>", childFieldGroup.getPath());
+        assertEquals("/testPathfoo<1>", childFieldGroup.getPath());
```

Comment:
I've changed the behavior here. When some item is removed after applying the filter, adjust succeeding indexes.